### PR TITLE
Configurable commands

### PR DESCRIPTION
**Pull request by disktnk**
refs #114

### CODE REVIEW

**Comment by disktnk** [comment]
I fixed test code not to depend on order of map keys.

File: cmd/build_sensorbee/main.go [hunk 1]
```diff
@@ -104,8 +108,12 @@ func loadConfig(path string) (*Config, error) {
 	}
 	// TODO: validation
 
-	config.SubCommands = []string{"run", "shell", "topology", "exp", "runfile"}
-	// TODO: sub commands should be configurable
+	if len(config.SubCommands) == 0 {
+		config.SubCommands = map[string]commandDetail{}
+		for _, sub := range sensorBeeDefaultCommands {
```
Comment (nobu-k) [inline cmd/build_sensorbee/main.go]: I think everybody knows that this file is `cmd/build_sensorbee/main.go`, so `defaultCommands` should be ok.

File: cmd/build_sensorbee/main_test.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+	}
+	if err := app.Run(os.Args); err != nil {
+		os.Exit(1)
+	}
+}
+`
+				So(string(b), ShouldEqual, expectedMainFile)
+			})
+		})
+	})
```
Comment (nobu-k) [inline cmd/build_sensorbee/main_test.go]: How about adding a test using multiple `SubCommands`? It doesn't have to perform an exact match of the contents but only needs to look up substrings like `repo1 "path/to/repo"` (possibly with regexp) to see if all subcommands are correctly imported.

**Comment by disktnk** [comment]
On travis-ci test, go1.4.3 is failed to build, but the failure is not related to this PR. Latest ugorji/go includes a function which is added from go1.5~ ( https://github.com/ugorji/go/commit/5099b68ed379e9b07d62052b23e326a7e95dbf75 ).

**Comment by nobu-k** [comment]
OK, let's set some official policy regarding which version of Go SensorBee supports. I think supporting latest three major versions (e.g. 1.7, 1.6, and 1.5 for now) is fine.

**Comment by nobu-k** [comment]
I mean, minor versions :)

**Comment by disktnk** [comment]
I think your proposal is good and will make issue about supported versions of Go, thanks.

File: cmd/build_sensorbee/main.go (hunk 1 above)
Comment (disktnk) [inline cmd/build_sensorbee/main.go]: renamed the variable

File: cmd/build_sensorbee/main_test.go (hunk 2 above)
Comment (disktnk) [inline cmd/build_sensorbee/main_test.go]: I separate to 2 test cases. 
1st., a created main file is exactly match or not when configuration includes a plugin and a command. 
2nd., a created main file has custom-command string or not with command configurations.

**Pull request merged**